### ISSUE TITLE
Command-line utility to add/remove files from an association

### ISSUE THREAD

**Issue by stscieisenhamer** [spacetelescope/jwst#2432]
# Issue
Data Science Processing (DSP), in order to handle [JSDP-688](https://jira.stsci.edu/browse/JSDP-688) and #2336, needs a CLI utility that will allow removal of files from an association.  Suggested API would be:

```
asn_edit [-r/-a] [-t exptype] [--ignore_suffix] [-o new_association_path] association.json filename [filename...]
```
where
- `-r`: remove listed files
- `-a`: add listed files
- `-t exptype`: If adding files, use specified `exptype`. Default is `science`
- `--ignore-suffix`: If removing files, first remove known JWST calibration suffixes from the file names when comparing against the files in the association.
- `-o`: If specified, create a new association file. Otherwise the input association file will be overwritten.

- `association.json` : The association to modify
- `filename [filename...]`: List of space-separated file names to add/remove from the association

**Comment by mswam** [comment]
looks good from the SDP point of view.  We can use this interface to remove a level3 input file from an association table that generated no level2b output.   The only trick seems to be determining what to enter for '-t exptype', but if 'science' is the right answer for these kinds of cases, that will work for us.

**Comment by stscieisenhamer** [comment]
@mswam For removal, the `-t` option will be unnecessary. So, removal would look like:
```
asn_edit -r asn.json expfile.fits
```
The `expfile.fits` would be removed and `asn.json` would be overwritten to reflect the change.

A wrinkle is what should you specify for `expfile.fits`? The Level3 will have the expected Level2b file in its list. However, the only thing you will know is that the Level2a file failed to produce. 

The "hard for SDP" option is to remove the 2a suffix, which will be `rate` or `rateints` and add on either the `cal` or `calints`.

The "hard for SCSB" option, but seems more practical, is to add another option, `--ignore_suffix`. Then, `asn_edit` would just take the exposure root name and do the removal based on that.

The command would then look like
```
asn_edit -r --ignore_suffix exposure_name_with_rate.fits level3_asn.json
```
and `asn_edit` would know to ignore the `_rate` and find the `_cal` version that will be in the association.

**Comment by mswam** [comment]
Sold!  I'd vote for the --ignore_suffix option.  That does keep it simpler, since different suffixes can show up in different situations.

**Comment by stscicrawford** [comment]
Closed with the merging of #2468
